app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -86,20 +96,80 @@ def proof_with_threatmetrix_if_needed(
         end
       end
 
-      def proof_resolution(applicant_pii:, timer:)
+      def proof_residential_address_if_needed(
+        applicant_pii:,
+        timer:,
+        double_address_verification:
+      )
+        return residential_address_unnecessary_result unless double_address_verification
+
+        timer.time('residential address') do
+          resolution_proofer.proof(applicant_pii)
+        end
+      end
+
+      def residential_address_unnecessary_result
+        Proofing::AddressResult.new(
+          success: true, errors: {}, exception: nil, vendor_name: 'ResidentialAddressNotRequired',
+        )
+      end
+
+      def resolution_unnecessary_result
+        Proofing::AddressResult.new(
+          success: true, errors: {}, exception: nil, vendor_name: 'ResolutionUnnecessary',
+        )
+      end
+
+      # TK: confirm vendor name with Team Ada
+      def resolution_cannot_pass
+        Proofing::AddressResult.new(
+          success: false, errors: {}, exception: nil, vendor_name: 'ResolutionCannotPass',
+        )
+      end
+
+      def proof_id_address_with_lexis_nexis_if_needed(applicant_pii:, timer:,
+                                                      residential_instant_verify_proof:)
+        return residential_instant_verify_proof if applicant_pii[:same_address_as_id] == 'true'
```

Comment:
This looks like it needs to check the `double_address_verification` value before deferring the result. Otherwise we may end up skipping this proofing step before DAV is enabled when users check the current `same_address_as_id` value.

Reply:
Added the `double_address_verification ` check.  I wrote a test for this bug, too.  Thanks again.